pkg/rbac/userData.go

```diff
@@ -192,7 +194,29 @@ func (user *userData) getNamespacedResources(cache *Cache, ctx context.Context,
 		}
 	}
 
+	//take intersection of authorizaed user namespaces and managed clusters:
+	managedClusters := cache.shared.managedClusters
+	namespaceNames := make([]string, len(nsResourcesAuth))
+	for name := range nsResourcesAuth {
+		namespaceNames = append(namespaceNames, name)
+	}
+
+	namespaceInt, err := intersection(namespaceNames, managedClusters)
```

Comment:
This should be stored in user's cache as a list to store the managed clusters that user can access.

Reply:
We should move it to a function on its own to populate user's managedClusters ([]string)

Reply:
Done.